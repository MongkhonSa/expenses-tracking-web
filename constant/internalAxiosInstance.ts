import axios from "axios";
import { resHandler, internalErrorHandler } from "./handler";

const internalAxiosInstance = axios.create({
  baseURL: "http://localhost:3000/api",
});

internalAxiosInstance.interceptors.response.use(
  resHandler,
  internalErrorHandler
);
export default internalAxiosInstance;
