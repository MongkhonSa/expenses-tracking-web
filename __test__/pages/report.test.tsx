import {
  fireEvent,
  getByRole,
  render,
  screen,
  waitFor,
} from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import internalAxiosInstance from "../../constant/internalAxiosInstance";

import ReportPage from "../../pages/report";
global.alert = jest.fn();
describe("Report Page", () => {
  const mockInternalAxiosInstance = jest.spyOn(internalAxiosInstance, "post");
  afterEach(() => {
    jest.clearAllMocks();
  });
  it("renders report page unchanged", () => {
    const { container, getByTestId } = render(<ReportPage />);
    mockInternalAxiosInstance.mockResolvedValue([
      { categoryName: "mockCategoryName1", total: 100 },
      { categoryName: "mockCategoryName2", total: 100 },
      { categoryName: "mockCategoryName3", total: 100 },
      { categoryName: "mockCategoryName4", total: 100 },
    ]);
    expect(container).toMatchSnapshot();
  });
  it("should call register api", async () => {
    mockInternalAxiosInstance.mockResolvedValue([
      { categoryName: "mockCategoryName", total: 100 },
    ]);

    const { getByTestId } = render(<ReportPage />);
    waitFor(() => {
      expect(screen.getByText("mockCategoryName")).toBeInTheDocument();
    });
    waitFor(() => {
      fireEvent.change(screen.getByRole("combobox"), {
        target: { value: "daily" },
      });
    });
    const dateInput = getByTestId("date-pick-test");
    fireEvent.mouseDown(dateInput);
    fireEvent.change(dateInput, { target: { value: "2022-08-27" } });
    fireEvent.click(document.querySelectorAll(".ant-picker-cell-selected")[0]);
    fireEvent.click(screen.getByText("expenses"));
  });
  it("should call register api and got error", async () => {
    mockInternalAxiosInstance.mockRejectedValue("mock");

    const { getByTestId } = render(<ReportPage />);
    waitFor(() => {
      expect(screen.getByText("mockCategoryName")).toBeInTheDocument();
    });
    waitFor(() => {
      fireEvent.change(screen.getByRole("combobox"), {
        target: { value: "daily" },
      });
    });
    const dateInput = getByTestId("date-pick-test");
    fireEvent.mouseDown(dateInput);
    fireEvent.change(dateInput, { target: { value: "2022-08-27" } });
    fireEvent.click(document.querySelectorAll(".ant-picker-cell-selected")[0]);
    fireEvent.click(screen.getByText("expenses"));
  });
});
