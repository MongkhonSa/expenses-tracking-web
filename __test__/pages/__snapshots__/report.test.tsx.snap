// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Report Page renders report page unchanged 1`] = `
<div>
  <div
    class="ant-row ant-row-end"
  >
    <div
      class="ant-col sc-724045b4-0"
    >
      <button
        class="ant-btn ant-btn-default"
        type="button"
      >
        <a
          href="/"
        >
          Home
        </a>
      </button>
    </div>
    <div
      class="ant-col sc-724045b4-0"
    >
      <button
        class="ant-btn ant-btn-default"
        type="button"
      >
        <a
          href="/report"
        >
          Report
        </a>
      </button>
    </div>
    <div
      class="ant-col sc-724045b4-0"
    >
      <button
        class="ant-btn ant-btn-default"
        type="button"
      >
        <h4>
          logout
        </h4>
      </button>
    </div>
  </div>
  <div
    class="ant-row ant-row-center"
  >
    <h1>
      Report
    </h1>
     
  </div>
  <div
    class="ant-row"
  >
    <div
      class="ant-col ant-col-6 ant-col-offset-2"
    >
      <div
        class="ant-select ant-select-single ant-select-show-arrow"
        date-testid="select-type"
        style="width: 100%;"
      >
        <div
          class="ant-select-selector"
        >
          <span
            class="ant-select-selection-search"
          >
            <input
              aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
              aria-autocomplete="list"
              aria-controls="rc_select_TEST_OR_SSR_list"
              aria-haspopup="listbox"
              aria-owns="rc_select_TEST_OR_SSR_list"
              autocomplete="off"
              class="ant-select-selection-search-input"
              id="rc_select_TEST_OR_SSR"
              readonly=""
              role="combobox"
              style="opacity: 0;"
              type="search"
              unselectable="on"
              value=""
            />
          </span>
          <span
            class="ant-select-selection-item"
            title="Daily"
          >
            Daily
          </span>
        </div>
        <span
          aria-hidden="true"
          class="ant-select-arrow"
          style="user-select: none;"
          unselectable="on"
        >
          <span
            aria-label="down"
            class="anticon anticon-down ant-select-suffix"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="down"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
              />
            </svg>
          </span>
        </span>
      </div>
    </div>
    <div
      class="ant-col ant-col-6 ant-col-offset-1"
    >
      <div
        class="ant-picker"
        style="width: 100%;"
      >
        <div
          class="ant-picker-input"
        >
          <input
            autocomplete="off"
            data-testid="date-pick-test"
            placeholder="Select date"
            readonly=""
            size="12"
            title=""
            value=""
          />
          <span
            class="ant-picker-suffix"
          >
            <span
              aria-label="calendar"
              class="anticon anticon-calendar"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="calendar"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                />
              </svg>
            </span>
          </span>
        </div>
      </div>
    </div>
    <div
      class="ant-col ant-col-6 ant-col-offset-1"
    >
      <div
        class="ant-radio-group ant-radio-group-outline"
        data-testid="type-group"
      >
        <div
          class="ant-row"
        >
          <label
            class="ant-radio-wrapper ant-radio-wrapper-checked"
          >
            <span
              class="ant-radio ant-radio-checked"
            >
              <input
                checked=""
                class="ant-radio-input"
                type="radio"
                value="income"
              />
              <span
                class="ant-radio-inner"
              />
            </span>
            <span>
              income
            </span>
          </label>
          <label
            class="ant-radio-wrapper"
          >
            <span
              class="ant-radio"
            >
              <input
                class="ant-radio-input"
                type="radio"
                value="expenses"
              />
              <span
                class="ant-radio-inner"
              />
            </span>
            <span>
              expenses
            </span>
          </label>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-row ant-row-center"
  >
    <div
      class="ant-col ant-col-20"
    >
      <div
        class="ant-table-wrapper"
      >
        <div
          class="ant-spin-nested-loading"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-table ant-table-empty"
            >
              <div
                class="ant-table-container"
              >
                <div
                  class="ant-table-content"
                >
                  <table
                    style="table-layout: auto;"
                  >
                    <colgroup />
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          class="ant-table-cell"
                        >
                          Category Name
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          Total
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        class="ant-table-placeholder"
                      >
                        <td
                          class="ant-table-cell"
                          colspan="2"
                        >
                          <div
                            class="ant-empty ant-empty-normal"
                          >
                            <div
                              class="ant-empty-image"
                            >
                              <svg
                                class="ant-empty-img-simple"
                                height="41"
                                viewBox="0 0 64 41"
                                width="64"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <g
                                  fill="none"
                                  fill-rule="evenodd"
                                  transform="translate(0 1)"
                                >
                                  <ellipse
                                    class="ant-empty-img-simple-ellipse"
                                    cx="32"
                                    cy="33"
                                    rx="32"
                                    ry="7"
                                  />
                                  <g
                                    class="ant-empty-img-simple-g"
                                    fill-rule="nonzero"
                                  >
                                    <path
                                      d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                                    />
                                    <path
                                      class="ant-empty-img-simple-path"
                                      d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                                    />
                                  </g>
                                </g>
                              </svg>
                            </div>
                            <div
                              class="ant-empty-description"
                            >
                              No Data
                            </div>
                          </div>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
